src/lib/nvme_pt_ioctl.c

```diff
@@ -2006,53 +2015,73 @@ int opal_set_pwd_pt(struct sed_device *dev, enum SED_AUTHORITY auth, const struc
 	return ret;
 }
 
-int opal_shadow_mbr_pt(struct sed_device *dev, const char *password,
-		uint8_t key_len, bool mbr)
+int opal_mbr_done_pt(struct sed_device *dev, const struct sed_key *key,
+		     bool mbr_done)
 {
-	struct sed_key disk_key;
-	int ret = 0;
-	uint8_t en_dis, enable_disable;
+	int ret;
+	uint8_t done;
 	struct opal_device *opal_dev;
 
-	if (password == NULL) {
-		SEDCLI_DEBUG_MSG("Need ADMIN1 password for mbr shadow "\
-				"enable/disable\n");
+	if (key == NULL) {
+		SEDCLI_DEBUG_MSG("User must provide ADMIN1 password\n");
 		return -EINVAL;
 	}
 
 	opal_dev = dev->priv;
 
-	if (mbr) {
-		enable_disable = OPAL_MBR_ENABLE;
-	} else {
-		enable_disable = OPAL_MBR_DISABLE;
-	}
+	if (mbr_done)
+		done = OPAL_TRUE;
+	else
+		done = OPAL_FALSE;
 
-	en_dis = (enable_disable == OPAL_MBR_ENABLE) ? OPAL_TRUE :
-		OPAL_FALSE;
+	ret = opal_start_admin1_lsp_session(dev->fd, opal_dev, key);
+        if (ret)
+                goto end_sessn;
 
-	ret = sed_key_init(&disk_key, password, key_len);
-	if (ret) {
-		return ret;
+        ret = opal_set_mbr_done(dev->fd, opal_dev, done);
+
+end_sessn:
+        opal_end_session(dev->fd, opal_dev);
+	return ret;
+}
+
+int opal_shadow_mbr_pt(struct sed_device *dev, const struct sed_key *key,
+			bool mbr)
+{
+	int ret = 0;
+	uint8_t enable_disable;
+	struct opal_device *opal_dev;
+
+	if (key == NULL) {
+		SEDCLI_DEBUG_MSG("Need ADMIN1 password for mbr shadow "\
+				"enable/disable\n");
+		return -EINVAL;
 	}
 
-	ret = opal_start_admin1_lsp_session(dev->fd, opal_dev, &disk_key);
+	opal_dev = dev->priv;
+
+	if (mbr)
```

Comment:
you could make it more pretty:)
done = mbr_done ? OPAL_TRUE : OPAL_FALSE;

Reply:
will do it